presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/SpillableHashAggregationBuilder.java

```diff
@@ -189,10 +189,10 @@ private boolean shouldMergeWithMemory(long memorySize)
     @Override
     public void close()
     {
-        if (hashAggregationBuilder != null) {
-            hashAggregationBuilder.close();
-        }
         try (Closer closer = Closer.create()) {
+            if (hashAggregationBuilder != null) {
```

Comment:
@nezihyigitbasi fair point. Indeed it is so, but it's not obvious. Since `close()` is quite sensitive operation, I'd prefer to err on the side of defensive programming. What do you think?